src/elfpy/pricing_models/hyperdrive.py

```diff
@@ -126,7 +126,7 @@ def calc_in_given_out(
             )
             market_result = MarketTradeResult(
                 d_base=-out.amount,
-                d_bonds=curve.market_result.d_bonds,
+                d_bonds=flat + curve.market_result.d_bonds,
```

Comment:
We don't want to add the flat component to the market bonds as this introduces a convexity problem. The previous implementation was correct.

Reply:
Ok thanks

Reply:
undid this.